tests/functional/z_cluster/nodes/test_nodes_maintenance.py

```
@@ -610,3 +624,277 @@ def test_pdb_check_simultaneous_node_drains(
+        log.info(
+            f"Eligible worker-only OSD nodes for shutdown: {shutdown_order} "
+            f"({eligible_count} of {total_hosts} total OSD hosts)"
+        )
+
+        # Phase 4: Gradual shutdown loop (Tiers 1-3)
+        # Tier 1 (full rebalance) is only reachable when total_hosts > size,
+        # i.e. there are spare nodes beyond what EC requires.
+        max_shutdowns = min(thresholds["min_drain_io_stops"], eligible_count)
+        spare_hosts = total_hosts - size
+        if spare_hosts > 0:
+            log.info(f"Tier 1 reachable: {spare_hosts} spare host(s) beyond k+m={size}")
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)